src/scripts/mod.rs

```diff
@@ -0,0 +1,318 @@
+
+pub fn init_scripts(scripts: String) -> Result<Vec<ScriptFile>> {
+    let mut scripts_to_run: Vec<ScriptFile> = Vec::new();
+
+    match scripts.as_ref() {
+        "none" => Ok(scripts_to_run),
+        "default" => {
+            let default_script = toml::from_str::<ScriptFile>(&DEFAULT).unwrap();
+            scripts_to_run.push(default_script);
+            Ok(scripts_to_run)
+        }
+        "custom" => {
+            let scripts_dir_base = match dirs::home_dir() {
+                Some(dir) => dir,
+                None => return Err(anyhow!("Could not infer scripts path.")),
+            };
+            let script_paths = match find_scripts(scripts_dir_base) {
+                Ok(script_paths) => script_paths,
+                Err(e) => return Err(anyhow!(e)),
+            };
+            debug!("Scripts paths \n{:?}", script_paths);
+
+            let parsed_scripts = parse_scripts(script_paths);
+            debug!("Scripts parsed \n{:?}", parsed_scripts);
+
+            let script_config = match ScriptConfig::read_config() {
+                Ok(script_config) => script_config,
+                Err(e) => return Err(anyhow!(e)),
+            };
+            debug!("Script config \n{:?}", script_config);
+
+            // Only Scripts that contain all the tags found in ScriptConfig will be selected.
+            if script_config.tags.is_some() {
+                let config_hashset: HashSet<String> =
+                    script_config.tags.unwrap().into_iter().collect();
+                for script in &parsed_scripts {
+                    if script.tags.is_some() {
+                        let script_hashset: HashSet<String> =
+                            script.tags.clone().unwrap().into_iter().collect();
+                        if config_hashset.is_subset(&script_hashset) {
+                            scripts_to_run.push(script.to_owned());
+                        }
+                        debug!(
+                            "\nScript tags does not match config tags {:?} {}",
+                            &script_hashset,
+                            script.path.clone().unwrap().display()
+                        );
+                    }
+                }
+            }
+            debug!("\nScript(s) to run {:?}", scripts_to_run);
+            Ok(scripts_to_run)
+        }
+        &_ => Ok(scripts_to_run),
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Script {
+    // Path to the script itself.
+    path: Option<PathBuf>,
+
+    // Ip got from scanner.
+    ip: IpAddr,
+
+    // Ports found with portscan.
+    ports: Vec<u16>,
+
+    // Port found in ScriptFile, if defined only this will run with the ip.
+    port: Option<String>,
+
+    // Character to join ports in case we want to use a string format of them, for example nmap -p.
+    join_ports: Option<String>,
+
+    // Tags found in ScriptFile.
+    tags: Option<Vec<String>>,
+
+    // The format how we want the script to run.
+    call_format: Option<String>,
```

Comment:
Shouldn't a script always have a `call_format`?

Reply:
I was thinking on some error handling logic after parsing the script, but you are right, it shouldn't be an `Option`, and we can rely on the toml parser to throw an error. Now we just don't run the scripts simply that's not conform in any way. But we can stop it at parsing.